### PR TITLE
Avoid blob url for files downloads

### PR DESCRIPTION
**Pull request by weizman**
* In order to allow the user to download log reports, we use a function called `exportAsFile`
* `exportAsFile` creates a URL object from a Blob and downloads that (common trick)
* The combination of URL object and Blob is dangerous in realms security and therefore is [forbidden by new Snow version](https://github.com/LavaMoat/snow/pull/69) because it's very easy to use it to bypass Snow
* In order to migrate into the new Snow version ([PR](https://github.com/MetaMask/metamask-extension/pull/17985)), I propose this change to the `exportAsFile` function, which will achieve the exact same goal but without the dangerous combo, but rather with `data:` URI instead
* (This modification is covered by tests in CI which proves it works 👍)

### CODE REVIEW

**Review by digiwand — DISMISSED** [review]

File: ui/helpers/utils/export-utils.js [hunk 1]
```diff
@@ -1,19 +1,23 @@
 import { getRandomFileName } from './util';
```
Comment (kumavis) [inline ui/helpers/utils/export-utils.js:2]: prefer async over callbacks

File: ui/helpers/utils/export-utils.js [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-  if (window.navigator.msSaveOrOpenBlob) {
-    window.navigator.msSaveBlob(blob, filename);
-  } else {
+  encodeB64(data, (b64) => {
+    // eslint-disable-next-line no-param-reassign
+    filename = filename || getRandomFileName();
     const elem = window.document.createElement('a');
-    elem.target = '_blank';
-    elem.href = window.URL.createObjectURL(blob);
+    elem.href = `data:${type};Base64,${b64}`;
```
Comment (kumavis) [inline ui/helpers/utils/export-utils.js:17]: can we just use `btoa`? https://developer.mozilla.org/en-US/docs/Web/API/btoa

Comment (weizman) [reply ui/helpers/utils/export-utils.js]: unfortunately `btoa` fails on special chars that we have in the exported json files that `TextEncoder` successfully deals with (I learned the hard way)

File: ui/helpers/utils/export-utils.js (hunk 1 above)
Comment (weizman) [reply ui/helpers/utils/export-utils.js:2]: 6de68ce1d29a66f8031fbcd65ac9f280debe6cf1

File: ui/helpers/utils/export-utils.js (hunk 2 above)
Comment (kumavis) [reply ui/helpers/utils/export-utils.js]: interesting -- what about `Buffer.from(xyz, 'utf8').toString('base64')`

Comment (weizman) [reply ui/helpers/utils/export-utils.js]: e95a668f0

File: ui/helpers/utils/export-utils.js [hunk 3]
```diff
@@ -1,19 +1,13 @@
 import { getRandomFileName } from './util';
 
-export function exportAsFile(filename, data, type = 'text/csv') {
+export async function exportAsFile(filename, data, type = 'text/csv') {
```
Comment (kumavis) [inline ui/helpers/utils/export-utils.js:3]: no longer needs to be async

Comment (weizman) [reply ui/helpers/utils/export-utils.js]: 6e13ea84285f0db5eb268ef52bab6a2b5cec49b3

**Review by kumavis — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Pull request merged**